extension/auth/configauth.go

```diff
@@ -0,0 +1,60 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auth // import "go.opentelemetry.io/collector/extension/auth"
```

Comment:
This file `configauth` should be kept in `configauth` package.

Reply:
I didn't get your point. Why this file should be kept in configauth package? In this way it means the auth.Settings should be defined out of the auth package? But I think the Settings are also part of auth package. 

Could you explain about this? Maybe my understanding is not correct.
@bogdandrutu

Reply:
Yes, the `auth.Settings` -> `configauth.Settings` anything else stays the same. This way "auth" contains only the component definition and "configauth" contains the config part that other components should use/embed.